Refactor FXIOS-8388 [v124] Move ActionFooterView to Fakespot code

bmihai23 opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8388)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18594)

## :bulb: Description
- This PR moves the ActionFooterView, currently part of the ComponentLibrary, to where it's used in Client/Frontend/Fakespot/Views. 
- This change improves code organization and avoids treating it as a separate component. 

### Changes Made
- Renamed ActionFooterView to FakespotActionFooterView
- Renamed ActionFooterViewModel to FakespotActionFooterViewModel
- Moved FakespotActionFooterView to Client/Frontend/Fakespot/Views
- Removed documentation for ActionFooterView from ComponentLibrary
- Removed ActionFooterComponentViewModel and ActionFooterViewController from SampleComponentLibraryApp

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

thatswinnie commented on firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift:
```diff
@@ -44,11 +45,10 @@ public final class ActionFooterView: UIView, ThemeApplicable {
     }
 
-    private lazy var linkButton: LegacyResizableButton = .build { button in
+    private lazy var linkButton: ResizableButton = .build { button in
```
This change plus actually causes UI issues where the button is either too big or displayed in with lots of space above. Let's undo this change and the removal of the buttonEdgeSpacing a few lines down and address the deprecation warning in another PR. I'll create a task for it.
![Simulator Screenshot - iPhone 15 Pro Max - 2024-02-07 at 09 35 26](https://github.com/mozilla-mobile/firefox-ios/assets/4530/7081009a-9685-499d-b176-2282ec53ae17)
![Simulator Screenshot - iPhone 15 Pro Max - 2024-02-07 at 09 35 18](https://github.com/mozilla-mobile/firefox-ios/assets/4530/a4b8e865-55a6-4fab-b610-d4e760ce52fe)

thatswinnie replied:
The task to fix the deprecation warnings for the button: https://github.com/mozilla-mobile/firefox-ios/issues/18634

thatswinnie reviewed (commented):
Thanks for grabbing this task @bmihai23 ! 🙏 

Unfortunately the SampleComponentLibraryApp doesn't build anymore because the files were not removed from the [SampleComponentLibraryApp Xcode project](https://github.com/mozilla-mobile/firefox-ios/tree/main/SampleComponentLibraryApp/SampleComponentLibraryApp.xcodeproj) correctly. Could you please get this fixed?

bmihai23 commented:
> Thanks for grabbing this task @bmihai23 ! 🙏
> 
> Unfortunately the SampleComponentLibraryApp doesn't build anymore because the files were not removed from the [SampleComponentLibraryApp Xcode project](https://github.com/mozilla-mobile/firefox-ios/tree/main/SampleComponentLibraryApp/SampleComponentLibraryApp.xcodeproj) correctly. Could you please get this fixed?

Thank you for your feedback, @thatswinnie! I've addressed the issue and fixed it.

thatswinnie reviewed (approved):
Perfect. Thank you so much @bmihai23 🙏
